api/app/utils/testing.py

```diff
@@ -9,8 +9,10 @@ def setUp(self):
         app.config['TESTING'] = True
         app.config['DEBUG'] = True
         self.app = app.test_client()
+        db.drop_all()
         db.create_all()
 
     def tearDown(self):
         db.session.remove()
         db.drop_all()
+        db.engine.execute('DELETE FROM alembic_version;')
```

Comment:
Can I get feedback on the changes I made here please since I'm not 100% sure if I was just missing something and this was just a hack?

The reason I do `db.drop_all()` in the `setUp()` method is because I want the db to start from a clean slate. I noticed that while dev'ing, if you run the tests, the state of the db is used as it already is making the tests slightly volatile.

The last line to clear the `alembic_version` table is to ensure that I can run the `upgrade` after running the tests. The `tearDown()` drops all the tables except the `alembic_version` which has the name of the last migration script that was run. If I run `upgrade`, it doesn't do anything then since it thinks all the tables are there but of course the tests dropped the tables.

Reply:
This is perfect